src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -153,7 +158,9 @@ def login(cmd, username=None, password=None, service_principal=None, tenant=None
 
     if service_principal:
         from azure.cli.core.auth.identity import ServicePrincipalAuth
-        password = ServicePrincipalAuth.build_credential(password, client_assertion, use_cert_sn_issuer)
+        password = ServicePrincipalAuth.build_credential(
+            secret_or_certificate=password, client_assertion=client_assertion,
+            certificate=certificate, use_cert_sn_issuer=use_cert_sn_issuer)
```

Comment:
Passing keyword arguments as positional arguments is fragile and may break unexpectedly.